tests/general/testdata/resource_metrics/incompat_env_config_source_labels.yaml

```diff
@@ -10,21 +10,21 @@ resource_metrics:
               state: used
               single-dollar: an-envvar-value-suffix
               single-dollar-no-curly-braces: prefix-an-envvar-value
-              double-dollar: ${env:AN_ENVVAR}-suffix
```

Comment:
Because of https://github.com/open-telemetry/opentelemetry-collector/pull/7055 I think this is now guaranteed to be expanded by the upstream provider. Existing compatibility persists in the non-`${env:..}` case.